### ISSUE TITLE
Add a dedicated SGLang model provider

### ISSUE THREAD

**Issue by virajmehta** [tensorzero/tensorzero#456]
> [!TIP]
>
> You can already use TensorZero + SGLang using the [OpenAI-compatible model provider](https://www.tensorzero.com/docs/gateway/guides/providers/openai-compatible).

---

Let's add a dedicated inference provider for SGLang (`provider_type = "sglang"`).

* Create a new provider in `gateway/src/inference/providers`
* Add the necessary entries to `gateway/src/model.rs` and `gateway/src/error.rs`
* Add your provider to the E2E test suite: `gateway/tests/e2e/providers`
* Add a minimal example to `examples/guides/providers`
* As the codebase evolves, providers are references elsewhere as well. Search for `mistral` and see wherever it is mentioned. Chances are there should be a similar mention of `sglang`.
 
See #529 and #547 for recent provider integration PRs.

**Comment by Gimmyalex** [comment]
Hi @GabrielBianconi & @virajmehta, can I work on this?

**Comment by GabrielBianconi** [comment]
Sounds good @Gimmyalex thanks!

**Comment by Gimmyalex** [comment]
Thanks, I'll get started on it and ask questions/guidance where necessary

**Comment by Gimmyalex** [comment]
Hey @GabrielBianconi, I'm having some issues with my sglang setup. I tried the docker image but my connection is poor.
I also tried colab notebook but keep getting an error with the server connected to the LLM I had set up (particularly gemma-2 for testing).
Any guidance on this matter is much appreciated.

**Comment by GabrielBianconi** [comment]
Hi @Gimmyalex - if I understand correctly, you're having issues with SGLang itself (independently of TensorZero)? I don't have experience with SGLang so unfortunately I can't help with that.

**Comment by Gimmyalex** [comment]
Yeah, this issue is with sglang, I successfully set up Tensorzero and have been using it for various tests, but sglang seems to be an issue for me.
I will continue doing some research as I had already written the code for the tests but haven't yet tested the sglang server

**Comment by Gimmyalex** [comment]
If it's okay, I would like to open an issue for the integration of NVIDIA NIM services for various LLMS as I'm familiar with it

**Comment by Gimmyalex** [comment]
Hi @GabrielBianconi, what's your take on this?

**Comment by GabrielBianconi** [comment]
Hi @Gimmyalex - our team can finish SGLang then. Feel free to work on NIM and send a PR, thanks!

**Comment by virajmehta** [comment]
Closed in #771
